plugins/modules/eos_user.py

```diff
@@ -233,6 +234,19 @@ def validate_privilege(value, module):
         )
 
 
+def get_os_version(module):
+    os_version = "4.20.10"
+    response = run_commands(
+        module, 'show version | grep "Software image version"'
+    )
+    version_match = re.search(
+        r"Software image version:\s+([\d\.]+)", response[0], re.M
+    )
+    if version_match:
+        os_version = float(version_match.group(1))
```

Comment:
So a couple of problems here. If this is a version number with three parts as the default indicates, the cast to float will fail, which is why I suggested a tuple of ints which will support any number of parts to the version.

The other issue is that both the default and the comparison value below are still strings, which are not comparable to floats, so if you ever do get the version out of this function, the comparison will fail with a TypeError